### ISSUE TITLE
ProxyClientSession options can be consolidated

### ISSUE THREAD

**Issue by d2r** [apache/trafficserver#3489]
Created as a follow up to a [review comment](https://github.com/apache/trafficserver/pull/3421/files#r179971092).

```C++
new_session->outbound_ip4       = options.outbound_ip4;
new_session->outbound_ip6       = options.outbound_ip6;
new_session->outbound_port      = options.outbound_port;
```

This could be logically grouped in an `options` member, such that the following would result:

```C++
new_session->options = options;
```

**Comment by d2r** [comment]
This is still a valid issue.

**Comment by bryancall** [comment]
This issue was fixed in this PR: #7584

**Comment by maskit** [comment]
It was fixed in #5852.
